src/services/ApiService.js

```diff
@@ -0,0 +1,73 @@
+import axios from "axios";
+
+export default class Api {
+  errMsg = "Coudn't fetch";
+
+  constructor(url) {
+    this.apibase = url;
+  }
+
+  msg(res) {
+    if (res.status === 200) return res.data;
+    throw new Error(`${this.errMsg} ${this.apibase}`);
+  }
+
+  async login(object) {
+    const res = await axios.post(`${this.apibase}/user/login`, object);
+    return this.msg(res);
+  }
+
+  async register(object) {
+    const res = await axios.post(`${this.apibase}/user/register`, object);
+    return this.msg(res);
+  }
+
+  async allVisits(token) {
```

Comment:
have you dynamic token ?

Reply:
I'm getting token after logging in /registering, so if I pass that token to constructor I got to initialize 
new Api class instance, also I can't go that way because I'm using Apicontext. So you say that I got to hoist token to the very top of the app after getting it from response?! That's not nice way I think, because then I'll need state for serving new token that has come. And by the way token is needed in methods excluding login and register (not everywhere).

Reply:
How can I get token if I'm not logged in or not registered? So If I go that way I must pass false / token to new Api or even must make a default token parameter in constructor with value of false constructor depending on situation. And I think that's totally disgusting